### DISCUSSION TITLE
curl.exe and 'CURL_WINDOWS_APP'

### DISCUSSION THREAD

**Discussion by gvanem** [curl/curl#11842] [Q&A] [answered]
Compiling `curl.exe` for *Windows Universal Windows Platform* (i.e. `CURL_WINDOWS_APP` gets defined),
gives me some errors in `src/tool_*.c` files. E.g.:
```c
tool_filetime.c(48,11): error: call to undeclared function 'CreateFile'; ISO C99 and later do not support implicit function declarations
      [-Wimplicit-function-declaration]
  hfile = CreateFile(tchar_filename, FILE_READ_ATTRIBUTES,
          ^
```
and:
```c
tool_doswin.c(637,15): error: call to undeclared function 'SearchPath'; ISO C99 and later do not support implicit function declarations
      [-Wimplicit-function-declaration]
    res_len = SearchPath(NULL, bundle_file, NULL, PATH_MAX, buf, &ptr);
              ^
tool_doswin.c(659,3): error: use of undeclared identifier 'MODULEENTRY32'
  MODULEENTRY32 mod = {0};
  ^
```

So what is the benefit of UWP wrt. to the `curl.exe` tool?

**Comment by vszakats** [comment]
How useful an UWP `curl.exe` is? probably not too much, unless maybe someone would distribute it via a channel requiring UWP.

`curl.exe` wasn't the only part that failed with UWP:
https://github.com/curl/curl/actions/runs/9358657250/job/25760914494

An attempt to make the build finish successfully:
https://github.com/curl/curl/pull/13870

~~It disables building `curl.exe` in UWP mode for now, but fixing it for UWP would probably be better if someone is interested in it.~~
The PR has changes that may or may not actually work, I cannot run UWP apps to test (nor does the CI, last time I checked).

**Comment by vszakats** [comment, marked as answer]
With #13870 merged, `curl.exe` builds again in UWP mode. Test binaries build too. Some things will most certainly not work as expected when run, though. Some feature, like those in the OP, are disabled now.

UWP is similar to WinCE, both are dead platforms, and both are apparently (lightly) supported by curl. Now UWP has a CI job to maintain that support.
